### PR TITLE
Feature/OINK-674, create deleteAccount endpoint.

### PR DESCRIPTION
**Pull request by pellerano**
- feat: added deleteAccount endpoint.

### CODE REVIEW

**Comment by juandatorr99** [comment]
Task linked: [OINK-674 Back-end delete account](https://app.clickup.com/t/24497898/OINK-674)

**Review by Royer99 — APPROVED** [review]
Bien

File: src/users/users.service.ts [hunk 1]
```diff
@@ -162,12 +160,11 @@ export class UsersService {
           'Transactions are still pending, please wait until all transactions are completed before deleting account.'
         );
 
-      // 3. Delete login data from firebase.
+      await this.authService.logout();
       await deleteUser(user);
+      await this.transactionsService.deleteAllUserTransactions(id);
+      await this.userRepository.delete({ id: id });
```
Comment (Topi99) [inline src/users/users.service.ts:166]: No se si sea buena idea borrar a los usuarios, tal vez podemos hacer un soft delete para evitar side effects.

Comment (juandatorr99) [reply src/users/users.service.ts:166]: Que side effects podrian existir?

**Review by juandatorr99 — APPROVED** [review]

**Pull request merged**